openedx/core/djangoapps/course_groups/tests/test_cohorts.py

```diff
@@ -172,6 +172,52 @@ def test_get_cohort_id(self):
             lambda: cohorts.get_cohort_id(user, SlashSeparatedCourseKey("course", "does_not", "exist"))
         )
 
+    def test_assignment_type(self):
+        """
+        Make sure that cohorts.set_assignment_type() and cohorts.get_assignment_type() works correctly.
+        """
+        course = modulestore().get_course(self.toy_course_key)
+        self.assertFalse(course.is_cohorted)
+
+        config_course_cohorts(course, discussions=[], cohorted=True)
+
+        # We are creating two random cohorts because we can't change assignment type of
+        # random cohort if it is the only random cohort present.
+        cohort1 = CohortFactory(course_id=course.id, name="TestCohort1")
+        CourseCohortFactory(course_user_group=cohort1, assignment_type=CourseCohort.RANDOM)
+        cohort2 = CohortFactory(course_id=course.id, name="TestCohort2")
+        CourseCohortFactory(course_user_group=cohort2, assignment_type=CourseCohort.RANDOM)
+        cohort3 = CohortFactory(course_id=course.id, name="TestCohort3")
+        CourseCohortFactory(course_user_group=cohort3, assignment_type=CourseCohort.MANUAL)
+
+        self.assertEqual(cohorts.get_assignment_type(cohort1), CourseCohort.RANDOM)
+
+        cohorts.set_assignment_type(cohort1, CourseCohort.MANUAL)
+        self.assertEqual(cohorts.get_assignment_type(cohort1), CourseCohort.MANUAL)
+
+        cohorts.set_assignment_type(cohort3, CourseCohort.RANDOM)
+        self.assertEqual(cohorts.get_assignment_type(cohort3), CourseCohort.RANDOM)
+
+    def test_cannot_set_assignment_type(self):
+        """
+        Make sure that we can't change the assignment type of a random cohort if it is the only random cohort present.
+        """
+        course = modulestore().get_course(self.toy_course_key)
+        self.assertFalse(course.is_cohorted)
+
+        config_course_cohorts(course, discussions=[], cohorted=True)
+
+        cohort = CohortFactory(course_id=course.id, name="TestCohort")
+        CourseCohortFactory(course_user_group=cohort, assignment_type=CourseCohort.RANDOM)
+
+        self.assertEqual(cohorts.get_assignment_type(cohort), CourseCohort.RANDOM)
+
+        exception_msg = "There must be one cohort to which students can be randomly assigned."
```

Comment:
maybe define this string as a constant in the cohorts module, and refer to it by the variable name here?